compiler/base.ts

```diff
@@ -107,6 +207,15 @@ export class LineCompiler implements InstrVisitor<void>, ExprVisitor<void> {
     this.routineType = routineType;
     this.isError = false;
     this.errors = [];
+
+    this.global = new GlobalBlock();
+    this.global.init(this, null, null, this.global);
```

Comment:
I think what would be safest is if global's parent is not the global block. This may make sense from the perspective of the language, but this is asking for problems when walking up the tree.